tests/pamiq_core/console/test_cui.py

```diff
@@ -0,0 +1,212 @@
+import json
+import re
+import sys
+from collections.abc import Generator
+from unittest.mock import MagicMock, patch
+
+import httpx
+import pytest
+
+from pamiq_core.console.cui import Console, main
+
+
+class TestConsole:
+    @pytest.fixture
+    def mock_httpx(self) -> Generator[MagicMock, None, None]:
+        # Mock httpx module imported in pamiq_core/console/cui.py
+        with patch("pamiq_core.console.cui.httpx") as mock_httpx:
+            # Mock GET response
+            mock_response = MagicMock()
+            mock_response.text = json.dumps({"status": "running"})
+            mock_httpx.get.return_value = mock_response
+            # Mock POST response
+            api_response = MagicMock()
+            api_response.text = json.dumps({"result": "success"})
+            mock_httpx.post.return_value = api_response
+            yield mock_httpx
```

Comment:
mockerフィクスチャを使う。あとMagicMockより純粋なMockの方が副作用が少なくてテストの健全性が高い。ただ、それでエラーが出た場合はMagicMockを用いて欲しい。
```suggestion
    def mock_httpx(self, mocker: MockerFixture) -> Generator[MagicMock, None, None]:
        # Mock httpx module imported in pamiq_core/console/cui.py
        mock_httpx = mocker.patch("pamiq_core.console.cui.httpx")
        # Mock GET response
        mock_response = mocker.Mock()
        mock_response.text = json.dumps({"status": "running"})
        mock_httpx.get.return_value = mock_response
        # Mock POST response
        api_response = mocker.Mock()
        api_response.text = json.dumps({"result": "success"})
        mock_httpx.post.return_value = api_response
        return mock_httpx
```